### ISSUE TITLE
Animation style gets broken

### ISSUE THREAD

**Issue by orianda** [gilbarbara/react-from-dom#21]
Using a node with a style attribute which itself contains an animate declaration then the spaces get removed and the animation does not work in the output:

Input:
```html
<g style="animation:aniname 50ms linear infinite normal forwards">...</g>
```

Output:
```html
<g style="animation:aninamemslinearinfinitenormalforwards">...</g>
```

The removement of all whitespaces in general might cause this behavior:
https://github.com/gilbarbara/react-from-dom/blob/66123d9e35c38d8f79b5f9326b7486a9df059e93/src/helpers.ts#L7

Maybe sth like:
```
import {camelCase} from 'lodash';
import type {CSSProperties} from 'react';

/**
 * @todo Is this issue fixed?
 * @see https://github.com/gilbarbara/react-from-dom/issues/21
 */

enum Mode {
  'preName',
  'name',
  'postName',
  'preValue',
  'value'
}

export const styleToObject = (
  input: string
): CSSProperties => {
  const style: Record<string, string | number | undefined> = {};
  let mode: Mode = Mode.preName;
  let skip = false;
  let name = '';
  let value = '';
  let quote = '';
  let escape = '';
  const append = (
    name: string,
    value: string
  ): void => {
    const key = name.startsWith('-')
      ? camelCase(`x${name}`).substring(1)
      : camelCase(name);
    const number = parseFloat(value);
    style[key] = isNaN(number) ? value : number;
  };
  const report = (
    index: number
  ): void => {
    skip = true;
    const error = new Error(`Unexpected character at position ${index} of "${input}"`);
    console.error(error);
  };
  `${input};`
    .split('')
    .map((
      char,
      index
    ) => {
      if (char === '\\' && !escape) {
        return char;
      }
      if (skip) {
        if (char === ';' && !escape) {
          mode = Mode.preName;
          name = '';
          value = '';
        } else {
          return;
        }
      }
      if (mode === Mode.preName) {
        if ((/\s/).test(char) || char === ';' && !escape) {
          return;
        } else if ((/[a-z\d-]/i).test(char)) {
          mode = Mode.name;
          name = char;
          return;
        } else {
          report(index);
          return;
        }
      }
      if (mode === Mode.name) {
        if ((/[a-z\d-]/i).test(char)) {
          name += escape + char;
          return;
        } else if ((/\s/).test(char) || char === ':' && !escape) {
          mode = Mode.postName;
        } else {
          report(index);
          return;
        }
      }
      if (mode === Mode.postName) {
        if ((/\s/).test(char)) {
          return;
        } else if (char === ':' && !escape) {
          mode = Mode.preValue;
          return;
        } else {
          report(index);
          return;
        }
      }
      if (mode === Mode.preValue) {
        if ((/\s/).test(char)) {
          return;
        } else if (char === ';' && !escape) {
          report(index);
          return;
        } else {
          mode = Mode.value;
          value = '';
        }
      }
      if (mode === Mode.value) {
        if (char === ';' && !escape && !quote) {
          append(name, value);
          mode = Mode.preName;
          name = '';
          value = '';
        } else {
          quote = escape
            ? quote
            : quote && quote === char
              ? ''
              : quote
                ? quote
                : char === '\'' || char === '"'
                  ? char
                  : '';
          value += escape + char;
          return;
        }
      }
      return;
    })
    .forEach((char: '\\' | '' = '') => {
      escape = char;
    });

  return style as CSSProperties;
};
```

**Comment by gilbarbara** [comment]
Hey @orianda 

I just needed to change one line! :)
Fixed in 0.7.3

**Comment by orianda** [comment]
This was fast. THX
